### PR TITLE
Repeat of Similarity explanations (`GradientSimilarity`) examples and documentation

### PR DESCRIPTION
**Pull request by ascillitoe**
This PR is a repeat of #650, with a bad merge commit (6530183d2475773a02791dedabe950e6c3ed8033) fixed. Refer to #650 for details of the original PR.

**Note:** I haven't updated licenses since the license file updates conflict with the current `feature/similarity-explainers` base.

### CODE REVIEW

File: doc/source/index.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ methods/CFRL.ipynb
 methods/IntegratedGradients.ipynb
 methods/KernelSHAP.ipynb
 methods/LinearityMeasure.ipynb
+methods/Similarity.ipynb
 methods/ProtoSelect.ipynb
```
Comment (jklaise) [inline doc/source/index.md:36]: Switch around to keep alphabetical order?

File: doc/source/index.md [hunk 2]
```diff
@@ -73,6 +74,9 @@ examples/integrated_gradients_imagenet.ipynb
 examples/integrated_gradients_mnist.ipynb
 examples/integrated_gradients_imdb.ipynb
 examples/integrated_gradients_transformers.ipynb
+examples/similarity_explanations_20ng.ipynb
+examples/similarity_explanations_mnist.ipynb
+examples/similarity_explanations_imagenet.ipynb
 examples/protoselect_adult_cifar10.ipynb
```
Comment (jklaise) [inline doc/source/index.md:80]: Switch around to keep alphabetical order? Can be delayed until #658.

File: doc/source/overview/getting_started.md [hunk 3]
```diff
@@ -161,7 +161,8 @@ alibi.explainers.__all__
 'plot_ale', 
 'IntegratedGradients', 
 'KernelShap', 
-'TreeShap']
+'TreeShap',
+'GradientSimilarity']
```
Comment (jklaise) [inline doc/source/overview/getting_started.md:165]: Switch around to keep alphabetical order? Can be delayed until #658.

**Review by jklaise — APPROVED** [review]
LGTM. Alphabetical order of docs links is minor and can be dealt with later, keen to merge.

**Review by gipster — COMMENTED** [review]
Looks the same as the version merged before. Nothing seems to be missing. It should be good to merge.

**Review by gipster — APPROVED** [review]

**Pull request merged**